agent/agent.go

```diff
@@ -807,7 +807,6 @@ func (a *Agent) flushLoop(
 			ticker.Reset()
 			logError(a.flushOnce(output, ticker, output.Write))
 		case <-output.BatchReady:
-			ticker.Reset()
 			logError(a.flushOnce(output, ticker, output.WriteBatch))
```

Comment:
It makes sense that if batches are continually becoming ready in less time than the write interval, it should not prevent the agent from writing all metrics. This change should fix that.

https://github.com/influxdata/telegraf/pull/8953 introduced this bug but also fixed https://github.com/influxdata/telegraf/issues/8941. I wonder if just removing the ticker.reset() will cause "did not complete within its flush interval" errors again.

We don't want to detect that the write interval is late just because we were writing a batch. Maybe we should pause the ticker before WriteBatch and unpause it after? The only ways to delay the ticker currently are the reset, stop, and next funcs. I guess we could add a pause func.

Reply:
I agree - this could re-introduce those messages. If we were to stop/reset the ticker we are back into the same situation we were in before. Pausing and start would eventually get us back to running the ticker, but in the worst case, where we are constantly doing batch writes, it could mean very long periods where we do not write out.

Do we know how prolific the error message was? I would rather that warning show up when there are long writes and prompt the user to increase the interval or decrease the batch size, than never write out. Thoughts?